src/main/java/cz/cvut/kbss/termit/persistence/dao/changetracking/ChangeTrackingContextResolver.java

```diff
@@ -39,8 +44,14 @@ public ChangeTrackingContextResolver(EntityManager em, Configuration config) {
     public URI resolveChangeTrackingContext(Asset<?> changedAsset) {
         Objects.requireNonNull(changedAsset);
         if (changedAsset instanceof Vocabulary) {
+            URI changeTrackingContextURI = resolveExistingChangeTrackingContext(changedAsset.getUri());
+            if(Objects.nonNull(changeTrackingContextURI))
+                return changeTrackingContextURI;
             return URI.create(changedAsset.getUri().toString().concat(contextExtension));
         } else if (changedAsset instanceof Term) {
+            URI changeTrackingContextURI = resolveExistingChangeTrackingContext(resolveTermVocabulary((Term) changedAsset));
```

Comment:
I suggest adding a method that would resolve change tracking context for the term in one query instead of first getting a vocabulary and then getting its change tracking context in a separate query.

Reply:
I don't think that is a good idea as it is a 2 step process with different outcomes. First, find the vocabulary and then resolve if it is in edited VC (if not, the `[vocabularyURI]/zmeny` is used). Second, If it is in an edited VC find the reference to ChangeTrackingContext (if not, throw an exception). 
If I would combine it into one query and it would not have a reference to ChangeTrackingContext or was not in VC the query would fail. Then I would still need to ask the DB what is the vocabulary and check if it is in the edited VCs. It would just transfer the double query problem from Assembly Line deployment to stand-alone deployment.

Reply:
Yeah, you are right about the types, I did not realize that.